### DISCUSSION TITLE
Question about $_comps

### DISCUSSION THREAD

**Discussion by ssbanerje** [ohmyzsh/ohmyzsh#10186] [Help] [answered]
When I use the `cargo` and `rustup` plugins, it seems that the assignment to `$_comps` made in the plugins [here](https://github.com/ohmyzsh/ohmyzsh/blob/735808f48d54aabce540f6c90294e21118104cf4/plugins/cargo/cargo.plugin.zsh#L10) and [here](
https://github.com/ohmyzsh/ohmyzsh/blob/735808f48d54aabce540f6c90294e21118104cf4/plugins/rustup/rustup.plugin.zsh#L11) throw an error that `$_comps` is not defined.

https://github.com/ohmyzsh/ohmyzsh/blob/735808f48d54aabce540f6c90294e21118104cf4/plugins/cargo/cargo.plugin.zsh#L10
https://github.com/ohmyzsh/ohmyzsh/blob/735808f48d54aabce540f6c90294e21118104cf4/plugins/rustup/rustup.plugin.zsh#L11

Here is the error:
```
oh-my-zsh/plugins/cargo/cargo.plugin.zsh:10: _comps: assignment to invalid subscript range
oh-my-zsh/plugins/rustup/rustup.plugin.zsh:11: _comps: assignment to invalid subscript range
```

I have been able to get around this by running the following before I source OMZ.
```zsh
declare -A _comps
```

I am not sure why this is happening. It seems like this variable should be set by default in ZSH.

I am using `zsh 5.8 (x86_64-apple-darwin20.1.0)` on MacOS BigSur

**Comment by mcornella** [comment, marked as answer]
This is a bug, probably due to using some kind of alternative to the `compinit` call that Oh My Zsh makes. I've opened https://github.com/ohmyzsh/ohmyzsh/issues/10190 to track this.
